R/input-file.R

```diff
@@ -101,6 +110,14 @@ fileInput <- function(inputId, label, multiple = FALSE, accept = NULL,
   if (length(accept) > 0)
     inputTag$attribs$accept <- paste(accept, collapse=',')
 
+  if (!is.null(capture)) {
+    if (tolower(capture) %in% c("user", "environment")) {
+      inputTag$attribs$capture <- tolower(capture)
+    } else {
+      # capture was previously a Boolean attribute which, if present, requested that the device's media capture device(s) such as camera or microphone be used instead of requesting a file input.
+      inputTag$attribs$capture <- NA
+    }
```

Comment:
```suggestion
    inputTag$attribs$capture <- switch(
      tolower(capture[1]),
      # Only standard values are `user` and `environment`
      "user" = "user",
      "environment" = "environment",
      # Missing attr value is allowed
      NA
    )
```

While the code above works, I'm trying to steer internal code to avoid %in% and use `switch` when possible.

Reprex:

``` r
bench_mark <- function(capture) {
  bench::mark(
    switch = {
      switch(
            tolower(capture[1]),
            # Only standard values are `user` and `environment`
            "user" = "user",
            "environment" = "environment",
            # Missing attr value is allowed
            NA
          )
    },
    ifin = {
      capture2 <- tolower(capture[1])
      if (capture2 %in% c("user", "environment")) {
        capture2
      } else {
        NA
      }
    }
  )
}

bench_mark(c("Environment", "wrong"))
#> # A tibble: 2 × 6
#>   expression      min   median `itr/sec` mem_alloc `gc/sec`
#>   <bch:expr> <bch:tm> <bch:tm>     <dbl> <bch:byt>    <dbl>
#> 1 switch       1.65µs   1.97µs   377007.        0B        0
#> 2 ifin         2.78µs   3.21µs   217854.        0B        0
bench_mark("user")
#> # A tibble: 2 × 6
#>   expression      min   median `itr/sec` mem_alloc `gc/sec`
#>   <bch:expr> <bch:tm> <bch:tm>     <dbl> <bch:byt>    <dbl>
#> 1 switch       1.54µs   1.86µs   468372.        0B        0
#> 2 ifin         2.74µs   3.18µs   283665.        0B        0
bench_mark(NA)
#> # A tibble: 2 × 6
#>   expression      min   median `itr/sec` mem_alloc `gc/sec`
#>   <bch:expr> <bch:tm> <bch:tm>     <dbl> <bch:byt>    <dbl>
#> 1 switch       1.28µs    1.5µs   587514.        0B     58.8
#> 2 ifin         2.29µs   2.65µs   349124.        0B      0
bench_mark("other")
#> # A tibble: 2 × 6
#>   expression      min   median `itr/sec` mem_alloc `gc/sec`
#>   <bch:expr> <bch:tm> <bch:tm>     <dbl> <bch:byt>    <dbl>
#> 1 switch        1.6µs   1.89µs   452467.        0B      0  
#> 2 ifin         2.68µs   3.12µs   298898.        0B     29.9
```

<sup>Created on 2021-08-13 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

Reply:
Dear @cpsievert, thanks for your response, but I don't think that it will work that way (at least the snippet that you shared) because the `fileInput` returning a complete block of HTML tags not a single tag.

Please note that tagAppendAttributes will append the attribute to the root tag in that block (i.e., in this case, the very first div tag `div class="form-group shiny-input-container"`)! Instead, we need that capture attribute much deeper in the `input` tag. 

Anyway, you may find a workaround solution I added in here #3483, which works well, but I believe that the Shiny framework should provide that fundamental attribute in an easier and straightforward way.